### ISSUE TITLE
SpEL referencing method parameter not working

### ISSUE THREAD

**Issue by spekdrum** [lukas-krecan/ShedLock#1284]
Since 0.17.0 version ShedLock supports SpEL, but this is not working:

```
@SchedulerLock(name = "{#hotelCode}", lockAtMostFor = "60s")
public void printMessage(final String hotelCode, final String message) {
...
}
```

This is what we get in Redis:
![imagen](https://user-images.githubusercontent.com/3244109/208083510-7b348817-b0b1-44e4-b300-4513d0e3e23b.png)

Also tried withoud curly braces "{ }"

**Comment by lukas-krecan** [comment]
Thanks for reporting. I will look into that.

**Comment by lukas-krecan** [comment]
Hi, this is surprisingly hard to implement. Have you seen any annotation other than `@Cacheable` supporting this?

**Comment by ipalbeniz** [comment]
I found an example in this article, maybe it will help you understand how it could be done: https://www.springcloud.io/post/2022-03/spel-aop-log/

**Comment by lukas-krecan** [comment]
Hi, I try to focus the library for the original purpose - scheduled task lock. I am afraid your use-case is out of the scope. That being said, you can use [this approach](https://github.com/lukas-krecan/ShedLock#locking-without-a-framework) which should solve your use-case.

**Comment by spekdrum** [comment]
Hi Lukas, I understand your point. IMHO the `@SchedulerLock` annotation is one of the key points of this library. It would be great to comply with this SpEL use case giving a wider support. The need to partition your locks can be a common requirement for big projects, for example, partition a lock per hotel for a method instead of locking for the entire hotel chain. I know this can be done programatically, as you pointed at the documentation, but if this can be done just using an annotation and SpEL would be great.

**Comment by lukas-krecan** [comment]
Hi, please correct me if I am wrong, but I do not think there is any SpEL spec of how to resolve method parameters. The only place I could find is the Cacheable annotation and the code seems to be really specific for the use-case of Caching.

Moreover, your use-case is clearly not scheduling but generic locking.

That being said, if you want to, you can submit a PR. If it does not bring too much complexity to the project, I will be happy to merge it. Otherwise, please implement it just for your project, Spring AOP is not that hard.

**Comment by Yannis-01** [comment]
This capability like java synchronized(object), synchronized(object) and synchronized method have different application scenarios. synchronized(object) is fine-grained

**Comment by lukas-krecan** [comment]
Released as 5.1.0
